beater/middleware/grpc_middleware.go

```diff
@@ -0,0 +1,66 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package middleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/elastic/apm-server/beater/request"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/monitoring"
+	"google.golang.org/grpc/codes"
+)
+
+// LogGRPC wraps implementations of gRPC methods to handle logging processing.
+func LogGRPC(ctx context.Context, logger *logp.Logger, f func() error) error {
+	var (
+		err   error
+		start = time.Now()
+	)
+
+	defer func() {
+		logger = logger.With("@timestamp", start, "event.duration", time.Since(start))
```

Comment:
I didn't see `@timestamp` specifically added elsewhere in the project, so I may be doing this unnecessarily / incorrectly